go/worker/storage/committee/node.go

```diff
@@ -665,21 +668,27 @@ func (n *Node) updateExternalServicePolicy(rtComputeNodes nodes.NodeDescriptorLo
 
 	executorCommitteePolicy.AddRulesForNodeRoles(&policy, rtComputeNodes.GetNodes(), node.RoleComputeWorker)
 
-	// TODO: Query registry only for storage nodes after
-	// https://github.com/oasisprotocol/oasis-core/issues/1923 is implemented.
-	nodes, err := n.commonNode.Consensus.Registry().GetNodes(n.ctx, consensus.HeightLatest)
-	if err != nil {
-		n.logger.Error("couldn't get nodes from registry", "err", err)
-	}
-	if len(nodes) > 0 {
-		// Only include storage nodes for our runtime.
-		var storageNodes []*node.Node
-		for _, nd := range nodes {
-			if nd.GetRuntime(n.commonNode.Runtime.ID()) != nil && nd.HasRoles(node.RoleStorageWorker) {
-				storageNodes = append(storageNodes, nd)
+	if n.rpcRoleProvider == nil {
```

Comment:
Change this to a `case` and add comments for both branches (e.g. when is the branch taken).